Tests/SpeziFoundationTests/RWLockTests.swift

```diff
@@ -77,7 +77,23 @@ final class RWLockTests: XCTestCase {
         wait(for: [expectation1, expectation2], timeout: 1.0)
     }
 
+    // This test is temporarily disabled on Linux.
+    //
+    // Reason: `lock.isWriteLocked()` behaves differently between Glibc (Linux) and macOS.
+    // On macOS, `pthread_rwlock_trywrlock()` returns `EDEADLK` when the calling thread
+    // already owns the lock, which makes `isWriteLocked()` work as expected.
+    //
+    // On Linux (glibc), `pthread_rwlock_trywrlock()` instead returns `EBUSY` in the same
+    // scenario (only `pthread_rwlock_wrlock()` can return `EDEADLK`)
+    //
+    // To make this portable, `RWLock` would need to explicitly track ownership
+    // of the lock by the current thread.
+    //
+    // See
+    // - https://linux.die.net/man/3/pthread_rwlock_trywrlock
+    // - https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/pthread_rwlock_trywrlock.3.html
```

Comment:
Should we create an issue for this to address it later on in follow-up works @phnagy?